etc/testing/deploy/aws.sh

```diff
@@ -5,46 +5,73 @@
 # kops clusters created for testing can always be enumerated and deleted.
 
 ## Parse command-line flags
-if [[ "$#" -lt 1 ]]; then
-  echo "Must pass --create, --delete, --delete-all or --list to testing/deploy/aws.sh"
-  exit 1
-fi
 
-REGION=us-west-1
-ZONE=us-west-1b
 STATE_STORE=s3://pachyderm-travis-state-store-v1
+REGION=-
+ZONE=us-west-1b
+OP=-
 
 # Process args
-new_opt="$( getopt --long="create,delete:,delete-all,list" -- ${0} "${@}" )"
+new_opt="$( getopt --long="create,delete:,delete-all,list,zone:" -- ${0} "${@}" )"
 [[ "$?" -eq 0 ]] || exit 1
 eval "set -- ${new_opt}"
 
-# No need to authenticate, as auth creds are already in environment variables
-# in travis
+while true; do
+  case "${1}" in
+    --delete-all)
+      OP=delete-all
+      shift
+      ;;
+    --list)
+      kops --state=${STATE_STORE} get clusters
+      exit 0  # Shortcut
+      ;;
+    --delete)
+      OP=delete
+      NAME="${2}"
+      shift 2
+      ;;
+    --create)
+      OP=create
+      shift
+      ;;
+    --zone)
+      ZONE="${2}"
+      shift 2
+      ;;
+    --)
+      shift
+      break
+      ;;
+  esac
+done
+
+len_zone_minus_one="$(( ${#ZONE} - 1 ))"
+REGION=${ZONE:0:${len_zone_minus_one}}
 
-case "${1}" in
-  --delete-all)
-    set -x
+echo -e "Region: ${REGION}\nZone: ${ZONE}"
+
+# No need to authenticate with kops, as auth creds are already in environment variables
+# in travis
+set -x
```

Comment:
Yup. This will capture the case statement, but it'll also print out the command that's actually run, which is the goal